ibm/resource_ibm_pi_network_port.go

```diff
@@ -0,0 +1,223 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	st "github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/helpers"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceIBMPINetworkPort() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceIBMPINetworkPortCreate,
+		Read:   resourceIBMPINetworkPortRead,
+		Update: resourceIBMPINetworkPortUpdate,
+		Delete: resourceIBMPINetworkPortDelete,
+		//Exists:   resourceIBMPINetworkExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			helpers.PINetworkName: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			helpers.PICloudInstanceId: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			/*
+			 "description": "",
+			    "ipAddress": "172.31.96.93",
+			    "macAddress": "fa:16:3e:30:b9:64",
+			    "portID": "6c9d0e42-73f3-492f-840d-3d4a7a573014",
+			*/
+
+			"ipaddress": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"macaddress": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"portid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ip_description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			//Computed Attributes
+
+		},
+	}
+}
+
+func resourceIBMPINetworkPortCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).IBMPISession()
+	if err != nil {
+		return err
+	}
+	powerinstanceid := d.Get(helpers.PICloudInstanceId).(string)
+	networkname := d.Get(helpers.PINetworkName).(string)
+
+	client := st.NewIBMPINetworkClient(sess, powerinstanceid)
+
+	networkPortResponse, err := client.CreatePort(networkname, powerinstanceid)
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Printing the networkresponse %+v", &networkPortResponse)
+
+	IBMPINetworkPortID := *networkPortResponse.PortID
+
+	d.SetId(fmt.Sprintf("%s/%s", powerinstanceid, IBMPINetworkPortID))
```

Comment:
To read the port we need three attributes powerinstanceid, n/wname, portID...so set the id a combination of all three and never use d.Get("****") in read methods